Varun/webscrape/oceannetwork

VaruneshBala opened the pull request:
# Description
BDTT-23 : Link [here](https://bottlenose.atlassian.net/jira/software/projects/BDTT/boards/1/backlog?selectedIssue=BDTT-23)
Fixes [Jira ticket #]()
<!-- Add Screenshots if there are changes or additions in frontend components -->

List all acceptance criterias of user story and indicate which has been solved previously and which is solved currently

Container Number field must be alphanumeric and it must not be empty [SOLVED]

After Container Number has been provided AND status found, ETA and LastUpdated timestamp must be displayed [SOLVED]

If there is no status found after providing container number, “No Status Found“ message must be displayed [SOLVED]
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ * ] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Migration

# Checklist:

- [ * ] I have added the appropriate labels to my PR
- [ * ] My code follows the style guidelines of this project
- [ * ] My changes generate no new warnings
- [ NA ] I have added tests that prove my fix is effective or that my feature works
- [ NA ] New and existing unit tests pass locally with my changes

pandaponn commented on app/webscrape/oceanContainerNum.py:
```diff
@@ -0,0 +1,58 @@
+
+    driver.maximize_window()
+
+    driver.get('https://ecomm.one-line.com/one-ecom/manage-shipment/cargo-tracking?ctrack-field=' + containerNum + '&trakNoParam=' + containerNum)
```
I understand that in testing you were using container and BL number. However, as this script accepts both container and BL number, and does not differentiate between the two in actual practice, we should assign a more general variable name instead of `containerNum`, such as `identifier`.

VaruneshBala replied:
changed to identifier!

pandaponn commented on app/webscrape/oceanContainerNum.py:
```diff
@@ -0,0 +1,58 @@
+from selenium.webdriver.common.by import By
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.chrome.options import Options
+from fake_useragent import UserAgent
```
Remove unused imports.

pandaponn reviewed (changes requested):
Great work picking up Selenium and completing this script in such a short amount of time @VaruneshBala! Appreciate the quick changes you have made as well since our call yesterday. If you have the time, it would be great if you could add in comments as well so that it is easier to dissect and understand the code.

VaruneshBala commented:
thank you!

made all the changes as requested please review

pandaponn commented:
@VaruneshBala the script fails to find the element on line 30

VaruneshBala commented:
fixed the webscrape for ONE

apparently there used to be a popup and i had a line of code to close the pop up. i think they removed the popup or smth and i removed the line of code so it seems to be working now

pandaponn reviewed (changes requested):
Please re-write the scraper as a flask app. You can follow ymluScraper.py

VaruneshBala commented:
rewrote scraper as a flask file. do check the scraper out and it works on my end when testing via postman

prussianman reviewed (changes requested):
Kindly include logic stated on site, i.e. to remove prefix "ONEY" in query BL before carrying out scraping

<img width="847" alt="image" src="https://user-images.githubusercontent.com/46068652/215296640-611ec80b-fce3-44b5-8bf4-150e7d550b6b.png">

prussianman reviewed (changes requested):
The previous change request will introduce some differences in the request handling between BL and CTR Numbers, hence, introducing different app routes for BL & CTR would be good. In addition, it adds standardisation between scrapers.

prussianman commented:
Tested with:

http://127.0.0.1:8084/ONE/SINC72210300 (excluded 'ONEY')
http://127.0.0.1:8084/ONE/TCNU7130634

VaruneshBala commented:
included the logic to remove ONEY prefix before webscraping. works on my end pls help me check

prussianman reviewed (approved):
Works, thanks!
